connection.js

```diff
@@ -419,19 +468,68 @@ class Connection {
             return { reject: true, msg: "Maximum handicap for unranked games is " + config.maxhandicapunranked + " , please increase the number of handicap stones" };
         }
 
-        if (!config.allowed_komi[notification.komi] && !config.allow_all_komi) {
-            conn_log("komi value " + notification.komi + " is not an allowed komi, allowed komi are: " + config.komi + ", rejecting challenge");
-            return { reject: true, msg: "komi " + notification.komi + " is not an allowed komi, please choose one of these allowed komi : " + config.komi };
+        if (!config.allowed_komi[notification.komi] && !config.allow_all_komi && !config.komiranked && !config.komiunranked) {
+            let notificationKomiString = "";
+            if (notification.komi === "null") {
+                notificationKomiString = "automatic";
+            } else {
+                notificationKomiString = notification.komi;
+            }
+            conn_log("komi value " + notificationKomiString + " is not allowed, allowed komi are: " + config.komi);
+            return { reject: true, msg: "komi " + notificationKomiString + " is not allowed, please choose one of these allowed komi : " + config.komi};
+        }
+
+        if (!config.allowed_komi_ranked[notification.komi] && notification.ranked && !config.allow_all_komi_ranked && config.komiranked) {
+            let notificationKomiString = "";
+            if (notification.komi === "null") {
+                notificationKomiString = "automatic";
+            } else {
+                notificationKomiString = notification.komi;
+            }
+            conn_log("komi value " + notificationKomiString + " is not allowed for ranked games, allowed komi for ranked games are: " + config.komiranked);
+            return { reject: true, msg: "komi " + notificationKomiString + " is not allowed for ranked games, please choose one of these allowed komi for ranked games: " + config.komiranked};
         }
 
-        if (!config.allowed_speeds[t.speed]) {
+        if (!config.allowed_komi_unranked[notification.komi] && !notification.ranked && !config.allow_all_komi_unranked && config.komiunranked) {
+            let notificationKomiString = "";
+            if (notification.komi === "null") {
```

Comment:
And another: `notification.komi === null`

Reply:
yes done, indeed, ty noticing it @Dorus :1st_place_medal: